lib/tests/fixtures/cpuid_fixtures.hpp

```diff
@@ -0,0 +1,42 @@
+#include "../fixtures.hpp"
+#include <internal/sources/cpuid_source.hpp>
+
+namespace whereami { namespace testing { namespace cpuid {
+
+    /**
+     * CPUID data source returning no usable information
+     */
+    class cpuid_fixture_empty : public sources::cpuid_base {
+    public:
+        sources::cpuid_registers read_cpuid(unsigned int leaf) const override;
+    };
+
+    /**
+     * CPUID data source with predefined register results
+     */
+    class cpuid_fixture_values : public sources::cpuid_base {
+    public:
+        cpuid_fixture_values(sources::cpuid_registers values);
```

Comment:
should this instead take a `map` of leaf -> registers, so you can override multiple queries?